Deploy MongoDB on Google Axion C4A virtual machine (#11)
odidev/arm-learning-paths: content/learning-paths/servers-and-cloud-computing/mongodb-on-gcp/baseline-testing.md

  },
  {
    _id: ObjectId('6892dacbbd44e23df474e39b'),
    record: 1,
    status: 'new',
    timestamp: ISODate('2025-08-06T04:32:11.101Z')
  },
  {
    _id: ObjectId('6892dacbbd44e23df474e39c'),
    record: 2,
    status: 'new',
    timestamp: ISODate('2025-08-06T04:32:11.103Z')
  },
  {
    _id: ObjectId('6892dacbbd44e23df474e39d'),
    record: 3,
    status: 'new',
    timestamp: ISODate('2025-08-06T04:32:11.104Z')
  },
  {
    _id: ObjectId('6892dacbbd44e23df474e39e'),
    record: 4,
    status: 'new',
    timestamp: ISODate('2025-08-06T04:32:11.106Z')
  }
]
```
5. Update a Document:

Update a specific document's field to validate update capability.
```javascript
db.test.updateOne({ record: 100 }, { $set: { status: "processed" } })
```
You should see an output similar to:

```output
{
  acknowledged: true,
  insertedId: null,
  matchedCount: 1,
  modifiedCount: 1,
  upsertedCount: 0
}
```
6. Delete a Document:

```javascript
db.test.deleteOne({ record: 100 })
```
Updates the status of record 100.

You should see an output similar to:

```output
{ acknowledged: true, deletedCount: 1 }
baselineDB> db.test.countDocuments()
9999
```

> 

Reviewer: **Suggestion** : Since record 100 is deleted, try display details of record=100, and it should not print anything...

7. Count Total Documents:

Count total entries to confirm expected data volume.
```javascript
db.test.countDocuments()
```

8. Measure Execution Time (Optional):

> Measure how long it takes to insert documents for performance insight.

Reviewer: ```suggestion
The below snippet measures how long it takes to insert documents for performance insight.
```

```javascript
var start = new Date()
for (let i = 0; i < 10000; i++) {
  db.test.insertOne({ sample: i })
}
print("Insert duration (ms):", new Date() - start)
```
You should see an output similar to:

```output
... for (let i = 0; i < 10000; i++) {
...   db.test.insertOne({ sample: i })
... }
... print("Insert duration (ms):", new Date() - start)
Insert duration (ms): 4427
```
9. Clean Up (Optional):

Deletes the **baselineDB** database and all its contents.
```javascript
db.dropDatabase()
```

> The above operations confirm that MongoDB is installed successfully and is functioning as expected on the GCP Arm64 environment. 

Reviewer: ```suggestion
The above is a destructive command that completely deletes the current database you are connected to in mongosh.

The above operations confirm that MongoDB is installed successfully and is functioning as expected on the GCP Arm64 environment. 
```

```output
{
  _id: ObjectId('689490ddb7235c65ca74e3fe'),
  record: 100,
  status: 'processed',
  timestamp: ISODate('2025-08-07T11:41:17.508Z')
}
```

7. Delete a Document:

```javascript
db.test.deleteOne({ record: 100 })
```
This tells MongoDB to delete one document from the test collection, where record is exactly 100.

You should see an output similar to:

```output
{ acknowledged: true, deletedCount: 1 }
```
Now, confirm the deletion:

```console
db.test.findOne({ record: 100 })
```
The above command confirms that the document was successfully deleted.

You should see an output similar to:
```output
null
```

8. Measure Execution Time (Optional):

The below snippet measures how long it takes to insert documents for performance insight.
```javascript
var start = new Date()
for (let i = 0; i < 10000; i++) {
  db.test.insertOne({ sample: i })
}
print("Insert duration (ms):", new Date() - start)
```
You should see an output similar to:

```output
Insert duration (ms): 4427
```
9. Count Total Documents:

Count total entries to confirm expected data volume.
```javascript
db.test.countDocuments()
```
You should see an output similar to:

```output
19999
```

> The count **19999** reflects the total documents after inserting 10,000 initial records, adding 10,000 more, and deleting one (record: 100).

Reviewer: ```suggestion
The count **19999** reflects the total documents after inserting 10,000 initial records, adding 10,000 more (in point 8), and deleting one (record: 100).
```
